From: Thomas Gleixner <tglx@linutronix.de>
Subject: ftrace: fix erroneous histogram stop WARN_ON() messages
Date: Wed, 28 Jan 2009 14:02:06 +0100

IRQs are temporarily disabled during histogram calculation in
cycles_2_ns(). This causes a nested irqs_off histogram processing
which emittes an erroneous WARN_ON() message and also confuses the
histogramm output.

Change local_irq to raw_local_irq in cycles_2_ns to prevent irqs_off
histogram processing.

Tested-by: Carsten Emde <C.Emde@osadl.org>
Signed-off-by: Thomas Gleixner <tglx@linutronix.de>

---
 include/asm-x86/timer.h |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: linux-2.6.26.8-rt14/include/asm-x86/timer.h
===================================================================
--- linux-2.6.26.8-rt14.orig/include/asm-x86/timer.h	2009-01-28 22:57:14.000000000 -0500
+++ linux-2.6.26.8-rt14/include/asm-x86/timer.h	2009-01-28 22:59:37.000000000 -0500
@@ -53,9 +53,9 @@ static inline notrace unsigned long long
 	unsigned long long ns;
 	unsigned long flags;
 
-	local_irq_save(flags);
+	raw_local_irq_save(flags);
 	ns = __cycles_2_ns(cyc);
-	local_irq_restore(flags);
+	raw_local_irq_restore(flags);
 
 	return ns;
 }
